### PR TITLE
Update GasketData type

### PR DESCRIPTION
**Pull request by egilchrist-godaddy**
## Summary

Properly define gasket data to be an object containing a config key which contains string or number keys that are potentially undefined.

## Changelog

- Update `GasketData` type to contain a generic object definition

## Test Plan

N/A. Ran linter and typecheck script to confirm no issues.

### CODE REVIEW

**Review by mmason2-godaddy — APPROVED** [review]

**Comment by mmason2-godaddy** [comment]
@egilchrist-godaddy Thanks for the contribution! Seems that tests are passing in the CI/CD. Give that a look:

```
@gasket/typescript-tests: Summary of all failing tests
@gasket/typescript-tests:  FAIL  test/nextjs.spec.tsx
@gasket/typescript-tests:   ● Test suite failed to run
@gasket/typescript-tests:     test/nextjs.spec.tsx:34:25 - error TS2741: Property 'config' is missing in type '{}' but required in type 'GasketData'.
@gasket/typescript-tests:     34       <GasketDataScript data={{}}/>;
@gasket/typescript-tests:                                ~~~~
@gasket/typescript-tests:       ../gasket-data/lib/index.d.ts:3:7
@gasket/typescript-tests:         3       config: { [key: string | number]: string | undefined };
@gasket/typescript-tests:                 ~~~~~~
@gasket/typescript-tests:         'config' is declared here.
@gasket/typescript-tests:       ../gasket-nextjs/src/index.d.ts:13:48
@gasket/typescript-tests:         13 export const GasketDataScript: ComponentType<{ data: GasketData }>;
@gasket/typescript-tests:                                                           ~~~~
@gasket/typescript-tests:         The expected type comes from property 'data' which is declared here on type 'IntrinsicAttributes & { data: GasketData; } & { children?: ReactNode; }'
@gasket/typescript-tests:     test/nextjs.spec.tsx:37:9 - error TS[2353](https://github.com/godaddy/gasket/actions/runs/8790070349/job/24123634418?pr=732#step:6:2354): Object literal may only specify known properties, and 'intl' does not exist in type 'GasketData'.
@gasket/typescript-tests:     37         intl: {
@gasket/typescript-tests:                ~~~~
@gasket/typescript-tests:       ../gasket-nextjs/src/index.d.ts:13:48
@gasket/typescript-tests:         13 export const GasketDataScript: ComponentType<{ data: GasketData }>;
@gasket/typescript-tests:                                                           ~~~~
@gasket/typescript-tests:         The expected type comes from property 'data' which is declared here on type 'IntrinsicAttributes & { data: GasketData; } & { children?: ReactNode; }'
@gasket/typescript-tests:     test/nextjs.spec.tsx:51:29 - error TS2741: Property 'config' is missing in type '{}' but required in type 'GasketData'.
@gasket/typescript-tests:     51         <GasketDataProvider gasketData={{}}>
@gasket/typescript-tests:                                    ~~~~~~~~~~
@gasket/typescript-tests:       ../gasket-data/lib/index.d.ts:3:7
@gasket/typescript-tests:         3       config: { [key: string | number]: string | undefined };
@gasket/typescript-tests:                 ~~~~~~
@gasket/typescript-tests:         'config' is declared here.
@gasket/typescript-tests:       ../gasket-nextjs/src/index.d.ts:40:68
@gasket/typescript-tests:         40 export const GasketDataProvider: ComponentType<PropsWithChildren<{ gasketData: GasketData }>>;
@gasket/typescript-tests:                                                                               ~~~~~~~~~~
@gasket/typescript-tests:         The expected type comes from property 'gasketData' which is declared here on type 'IntrinsicAttributes & { gasketData: GasketData; } & { children?: ReactNode; }'
@gasket/typescript-tests:     test/nextjs.spec.tsx:56:11 - error TS2353: Object literal may only specify known properties, and 'intl' does not exist in type 'GasketData'.
@gasket/typescript-tests:     56           intl: {
@gasket/typescript-tests:                  ~~~~
@gasket/typescript-tests:       ../gasket-nextjs/src/index.d.ts:40:68
@gasket/typescript-tests:         40 export const GasketDataProvider: ComponentType<PropsWithChildren<{ gasketData: GasketData }>>;
@gasket/typescript-tests:                                                                               ~~~~~~~~~~
@gasket/typescript-tests:         The expected type comes from property 'gasketData' which is declared here on type 'IntrinsicAttributes & { gasketData: GasketData; } & { children?: ReactNode; }'
@gasket/typescript-tests: Test Suites: 1 failed, 33 passed, 34 total
@gasket/typescript-tests: Tests:       116 passed, 116 total
@gasket/typescript-tests: Snapshots:   0 total
@gasket/typescript-tests: Time:        8.156 s
@gasket/typescript-tests: Ran all test suites.
@gasket/typescript-tests: npm ERR! Lifecycle script `test` failed with error: 
@gasket/typescript-tests: npm ERR! Error: command failed 
@gasket/typescript-tests: npm ERR!   in workspace: @gasket/typescript-tests@6.46.7 
@gasket/typescript-tests: npm ERR!   at location: /home/runner/work/gasket/gasket/packages/gasket-typescript-tests 

 

 >  Lerna (powered by Nx)   Running target test for 45 projects failed

   Failed tasks:
   
   - @gasket/typescript-tests:test

Error: Process completed with exit code 1.
```

File: packages/gasket-data/lib/index.d.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
 declare module '@gasket/data' {
-    export interface GasketData {}
+    export interface GasketData {
+      config: { [key: string | number]: string | undefined };
```
Comment (agerard-godaddy) [inline packages/gasket-data/lib/index.d.ts:3]: Although not the tightest type, this is already declared by the plugin that add the config prop [here](https://github.com/godaddy/gasket/blob/deec24a99256e6b875a08ec20ed83b40212a72e4/packages/gasket-plugin-config/lib/plugin.d.ts#L20-L22).

In your code, that expect the config plugin to have added this property, you can pull in the types by add towards the top of your file:
```
/// <reference types="@gasket/plugin-response-data"/>
```

**Comment by egilchrist-godaddy** [comment]
@mmason2-godaddy I've fixed the tests thank you for catching that!

File: packages/gasket-data/lib/index.d.ts (hunk 1 above)
Comment (egilchrist-godaddy) [reply packages/gasket-data/lib/index.d.ts]: I think I might've taken the wrong point from your feedback here, but I updated the gasket plugin config type as well 😛

**Review by agerard-godaddy — APPROVED** [review]

**Comment by egilchrist-godaddy** [comment]
@agerard-godaddy I don't have permission to merge. I just updated the branch with the latest changes from main again so feel free to merge whenever you can.

**Review by kbader-godaddy — APPROVED** [review]

**Pull request merged**